src/core/ControllerConnection.cpp

```diff
@@ -186,8 +189,11 @@ void ControllerConnection::saveSettings( QDomDocument & _doc, QDomElement & _thi
 		}
 		else
 		{
-			int id = Engine::getSong()->controllers().indexOf( m_controller );
-			if( id >= 0 )
+			auto& controllers = Engine::getSong()->controllers();
+			auto it = std::find(controllers.begin(), controllers.end(), m_controller);
+			int id = std::distance(controllers.begin(), it);
+
+			if (id >= 0)
```

Comment:
If `m_controller` is not found in `controllers`, `id` will be -1 in the previous code, but now it wouldn't.